crates/client/rpc/src/lib.rs

```diff
@@ -528,9 +528,17 @@ where
         request: Vec<BroadcastedTransaction>,
         block_id: BlockId,
     ) -> RpcResult<Vec<FeeEstimate>> {
-        // TODO:
-        //      - modify BroadcastedTransaction to assert versions == "0x100000000000000000000000000000001"
-        //      - to ensure broadcasted query signatures aren't valid on mainnet
+        let smthng = request.iter().any(|tx| match tx {
+            BroadcastedTransaction::Invoke(BroadcastedInvokeTransaction::V0(tx_v0)) => tx_v0.is_query,
+            BroadcastedTransaction::Invoke(BroadcastedInvokeTransaction::V1(tx_v1)) => tx_v1.is_query,
+            BroadcastedTransaction::Declare(BroadcastedDeclareTransaction::V1(tx_v1)) => tx_v1.is_query,
+            BroadcastedTransaction::Declare(BroadcastedDeclareTransaction::V2(tx_v2)) => tx_v2.is_query,
+            BroadcastedTransaction::DeployAccount(deploy_tx) => deploy_tx.is_query,
+        });
+
+        if smthng {
+            return Err(StarknetRpcApiError::UnsupportedTxVersion.into());
+        }
```

Comment:
Why are those not supported for `estimate_fee`?

Reply:
What do you mean, isn't the point of the issue to block/prevent all the one that aren't query_version?

Reply:
These are the cases:
 - Deploying: `__validate_deploy__` should ensure tx version is  2**128 + 1
 - Invoke: `__validate_` should ensure tx version is  2**128 + 1
 - Declaring: `__validate_declare__` should ensure tx version is  2**128 + 2
 
I think both BroadcastedInvokeTransaction::V0 and BroadcastedDeclareTransaction::V1 shouldn't even be supported 
so the another error should be thrown for those?

Reply:
ref:
https://docs.starknet.io/documentation/starknet_versions/version_notes/#:~:text=New%20transaction%20version%20(version%201)%20for%20invoke%20and%20declare%20transactions%3A

Reply:
https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/Blocks/transactions/

Reply:
Still working on this, so just ignore it at this point :)

Reply:
GM @tdelabro , @apoorvsadana ,
As stated in the documentation 
https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/Blocks/transactions/
Invoke v0 and declared v1 are deprecated and will be removed.
Should those already return some different error?
Or should this be part of another issue for when they will effectively be removed?